plugins/recorded_future/komand_recorded_future/util/api.py

```diff
@@ -37,21 +50,114 @@ def _call_api(self, method: str, endpoint: str, params: dict = None, data: dict
+        :param filename: filename of the compressed Gzip
+        :return: The decoded GZIP contents in Base64 and the decompressed JSON representation of the XML file
+        """
+        try:
+            with gzip.open(filename, "rb") as file:
+                compressed_data = file.read()
+                try:
+                    parsed_data = RecordedFutureApi.decompress_gzip_to_dict(compressed_data)
+                except MemoryError:
+                    RecordedFutureApi.logger.info("Response is too large to read. Returning GZIP...")
+                    parsed_data = None
+                decoded_data = base64.b64encode(compressed_data).decode("utf-8")
+                return decoded_data, parsed_data
+        except (gzip.BadGzipFile, IOError) as exception:
+            raise PluginException(
+                cause="Error reading saved GZIP response file",
+                assistance="File may contain errors or be in an unexpected format",
+                data=exception,
+            )
+        except (B64EncodingError, UnicodeDecodeError):
+            raise PluginException(preset=PluginException.Preset.BASE64_DECODE)
+
+    def _call_api_stream_to_file(
+        self, method: str, endpoint: str, params: dict = None, data: dict = None, json: dict = None
+    ) -> Tuple[Any, Any]:
+        _url = self.base_url + endpoint
+        response = requests.request(
+            url=_url, method=method, params=params, data=data, json=json, headers=self.headers, stream=True
+        )
+        RecordedFutureApi.response_handler(response, _url)
+        try:
+            with NamedTemporaryFile(delete=False) as file:
+                temp_filename = file.name
+                for chunk in response.iter_content(chunk_size=8192):
+                    compressed_chunk = gzip.compress(chunk)
+                    file.write(compressed_chunk)
+                    # flush increases performance overhead but reduces risk of memory error
+                    file.flush()
+            return RecordedFutureApi.decompress_and_process_gzip(temp_filename)
+        except IOError as exception:
+            raise PluginException(
+                cause="Error writing response content to file",
+                assistance="File may contain errors or the disk may be full",
+                data=exception,
+            )
+        except MemoryError as exception:
+            raise PluginException(
+                cause="Memory Error writing response content to file",
+                assistance="Insufficient memory to process response content",
+                data=exception,
+            )
+        except ValueError as exception:
+            raise PluginException(
+                cause="Error processing response content when writing to file",
+                assistance="Response content may contain invalid data",
+                data=exception,
+            )
+        finally:
+            if os.path.exists(temp_filename):
```

Comment:
I think it would be worthwhile to update at least one of the actions that uses this util to parse and return the gzip to test that the error handling / or at least the happy path that we fall into this `finally` block and remove the file? Just thinking someone could easily remove this os.remove() and then we'd be leaving around these old files.